Main/Source/proto.cpp

```diff
@@ -38,6 +40,19 @@ character* protosystem::BalancedCreateMonster()
 
         if(!DataBase->IsAbstract && DataBase->CanBeGenerated)
         {
+          truth Allowed = false;
+          const fearray<int> &dlist = DataBase->AllowedDungeons;
```

Comment:
I think `dlist` could be renamed to something more descriptive, and to use CamelCase for consistency. E.g. `AllowedDungeons`.

Other than that, looks good!